feat: Complication ModularLarge

raphschwander opened the pull request:
Hey!

Have a look at my implementation of complication ModularLarge.

It still needs some testing, however I couldn't test it on my iWatch as I don't have the right provisioning profile, only in the simulator, which is not so great to test.

It also needs some tuning/ improvement, specially for deciding when the complication data should be extended/reloaded. 

Currently the first station returned by the method "getStations" is defined as the one appearing on the clockFace and the data are reloaded every 1.5 hour. To this regard, what is the exact criteria for the ordering of the returned stations from method "getStations" ? It seemed to me it was the last favorited station first, but I noticed it takes into account the distance as well. 

| Complication ModularLarge | Time Travel |
| --- | --- |
| ![simulator screen shot 13 oct 2015 11 27 57](https://cloud.githubusercontent.com/assets/11804885/10451079/4f956fd2-719e-11e5-97ef-698839f64840.png) | ![simulator screen shot 13 oct 2015 11 33 50](https://cloud.githubusercontent.com/assets/11804885/10451092/5d73cac2-719e-11e5-89f3-81544f95ca62.png) |

Display the next two departures from the closest/favorited station on the clockFace in ModularLarge family

Once the complication is set by the user, it allows him to see immediately the remaining time till the next departures
The user can also use time travel feature to look at the future departures.

Limitations
- The station displayed on the clockFace corresponds to the first station returned from the "getStations" method
- The method "getStations" is called multiple times
- So far, the timeline entries are not reloaded when the order of the returned stations change
- The timeline entries are reloaded every 1.5 hour arbitrarly
- The text format of the bus/train line and remaining time till departure don't give a clear indication of what they display
- The destination is truncated

chregu commented:
wow. cool. I'll check it out soon

getStations() by the way returns the nearest favorite if that is within 1km, otherwise it returns the nearest station.

I'll add some comments to the code

It currently has a merge conflict, can you rebase against develop?

chregu commented on Time for Coffee! WatchOS 2 App Extension/ComplicationController.swift:
```diff
@@ -0,0 +1,225 @@
+        let departureLine = departure.getLine()
+        let nextDepartureLine = nextDeparture?.getLine() ?? "-"
+
+        let nextDepartureDestination = nextDeparture?.getDestination() ?? "-"
```
You can use nextDeparture?.getDestination(station) to have the localName part first (if the city matches the station city), so that it's clearer on the watch

chregu commented on timeforcoffee.xcodeproj/project.pbxproj:
```diff
@@ -754,7 +757,7 @@
 						CreatedOnToolsVersion = 7.0;
-						DevelopmentTeam = DN76797UYF;
+						DevelopmentTeam = 6EMM22S2YK;
 						SystemCapabilities = {
```
can you please move the DevTeam back them and the PROVISIONING_PROFILE changes below as well

raphschwander replied:
Oh great ! I didn't notice this alternative method for getDestination. Sure
I ll do the changes and rebase against master. If you see anything else,
let me know.

2015-10-14 9:03 GMT+02:00 Christian Stocker notifications@github.com:

> In timeforcoffee.xcodeproj/project.pbxproj
> https://github.com/timeforcoffee/timeforcoffee-ios/pull/4#discussion_r41960777
> :
> 
> > @@ -754,7 +757,7 @@
> >                     };
> >                     EC46C5271BB5C78A00F2A963 = {
> >                         CreatedOnToolsVersion = 7.0;
> > -                       DevelopmentTeam = DN76797UYF;
> > -                       DevelopmentTeam = 6EMM22S2YK;
> >                       SystemCapabilities = {
> 
> can you please move the DevTeam back them and the PROVISIONING_PROFILE
> changes below as well
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/timeforcoffee/timeforcoffee-ios/pull/4/files#r41960777
> .

raphschwander committed:
feat: Local name of the departure first
Show the localName of the departure first (if the city matches the station city)

raphschwander committed:
feat: Set custom color for complication

raphschwander committed:
feat: Complication ModularLarge

Display the next two departures from the closest/favorited station on the clockFace in ModularLarge family

Once the complication is set by the user, it allows him to see immediately the remaining time till the next departures
The user can also use time travel feature to look at the future departures.

Limitations
- The station displayed on the clockFace corresponds to the first station returned from the "getStations" method
- The method "getStations" is called multiple times
- So far, the timeline entries are not reloaded when the order of the returned stations change
- The timeline entries are reloaded every 1.5 hour arbitrarly
- The text format of the bus/train line and remaining time till departure don't give a clear indication of what they display
- The destination is truncated

raphschwander committed:
chore: Move back DevelopmentTeam and PROVISIONING_PROFILE

raphschwander commented:
Should be good now :+1:

chregu commented:
Cool. It Works on my real watch. Great work. Will merge it soon and release a new beta.

raphschwander commented:
Awsome ! Can't wait to try it on my iWatch :)

2015-10-15 20:35 GMT+02:00 Christian Stocker notifications@github.com:

> Merged #4 https://github.com/timeforcoffee/timeforcoffee-ios/pull/4.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/timeforcoffee/timeforcoffee-ios/pull/4#event-436729983
> .
